test/e2e/internal/notation/registry.go

```diff
@@ -0,0 +1,102 @@
+package notation
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"sync"
+
+	oregistry "oras.land/oras-go/v2/registry"
+)
+
+var (
+	repoId   = 0
+	repoIdMu = sync.Mutex{}
+)
+
+const (
+	testRepo = "e2e"
+	testTag  = "v1"
+)
+
+type Registry struct {
+	Host     string
+	Username string
+	Password string
+}
+
+var TestRegistry = Registry{}
+
+type Artifact struct {
+	*Registry
+	Repo string
+	Tag  string
+}
+
+// GenArtifact generates a new image with a new repository.
+func GenArtifact() *Artifact {
```

Comment:
nit: code style: naming convention
```suggestion
func GenerateArtifact() *Artifact {
```

Reply:
Updated.